mos_tests/failover/python_tests/test_restarts.py

```diff
@@ -0,0 +1,210 @@
+            logger.info('disable all galera services with cmd {}'.format(cmd))
+            remote.check_call(cmd)
+
+            logger.info('wait until all mysql processes are stopped')
+            wait(is_mysql_stopped, timeout_seconds=3 * 60, sleep_seconds=5)
+
+            cmd = 'pcs resource enable clone_p_mysql'
+            logger.info('enable all galera services with cmd {}'.format(cmd))
+            remote.check_call(cmd)
+
+            logger.info('wait until all mysql processes started')
+            wait(is_mysql_started, timeout_seconds=3 * 60, sleep_seconds=5)
+
+        self.check_common_services()
+
+    def test_restart_rabbitmq_services_with_replicaton(self):
+        """ Restart all RabbitMQ services with data replication
+        Scenario
+            1. Login to the first Openstack controller node
+               and restart RabbitMQ service: service rabbitmq-server restart
+            2. Wait while RabbitMQ is successfully started on the controller
+            3. Repeat steps -1-2 for all controller nodes in your cluster
+            4. Execute 'rabbitmqctl cluster_status'
+               and verify that all RabbitMQ nodes are in the same cluster.
+
+        """
+
+        controllers = self.env.get_nodes_by_role('controller')
+
+        for controller in controllers:
+            with controller.ssh() as remote:
+                cmd = 'service rabbitmq-server restart'
+                logger.info('restart all rabbitmq services by: {}'.format(cmd))
+                result = remote.check_call(cmd)
+
+        def is_rabbit_alive():
+            with controllers[0].ssh() as remote:
+                # The following cmd will return non zero exit code
+                # if rabbit is dead
+                cmd = 'rabbitmqctl cluster_status'
+                result = remote.execute(cmd)
+                if not result['exit_code']:
+                    return True
+
+        logger.info('wait until all rabbits come alive')
+        wait(is_rabbit_alive, timeout_seconds=10 * 60, sleep_seconds=5)
+
+        logger.info('check that all rabbitmq nodes are in the same cluster')
+        with controllers[0].ssh() as remote:
+            cmd = 'rabbitmqctl cluster_status'
+            result = remote.execute(cmd)
+            assert not result['exit_code'], 'failed to read the cluster status'
+            # Now count the rabbit nodes in the output
+            # Simply will check lines of output
+            # Until the needed info is found
+            # It should be like this:
+            # {running_nodes,
+            # ['rabbit@node-1','rabbit@node-2','rabbit@node-3']}
+            rabbit_nodes = 0
+            p = re.compile("\{running_nodes,.*(\[.*\])\}")
+            for line in result['stdout']:
+                res = p.findall(line)
+                if res:
+                    rabbit_nodes = len(ast.literal_eval(res[0]))
+                    logger.info(
+                        'running rabbit nodes are {}'.format(line))
```

Comment:
May be better change text ro change to `'rabbit running nodes: "{}"'` or something other.
`[INFO] running rabbit nodes are {running_nodes,['rabbit@node-1','rabbit@node-2','rabbit@node-3']}` looks some messy